platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/SocketFactory.java

```diff
@@ -62,7 +62,15 @@ static void configureAndBind(
             // set the IP_TOS option
             serverSocket.setOption(java.net.StandardSocketOptions.IP_TOS, socketConfig.ipTos());
         }
-        final InetSocketAddress endpoint = new InetSocketAddress(InetAddress.getByAddress(ALL_INTERFACES), port);
+
+        // Determine interface to bind to.
+        final InetAddress bindInterface = socketConfig.bindInterfaceHostname().equals(SocketConfig.ALL_IPV4_INTERFACES)
+                ? InetAddress.getByAddress(ALL_INTERFACES)
+                : InetAddress.getByName(socketConfig.bindInterfaceHostname());
+        // Determine port to bind to.
+        final int bindPort = socketConfig.bindInterfacePort() < 0 ? port : socketConfig.bindInterfacePort();
```

Comment:
the fact that the input parameter `port` may or may not be used isn't very clear in this method, probably the variable should be renamed to something like `fallbackPort` and the javadoc updated